### PR TITLE
Implement ZIO#tapDefect

### PR DESCRIPTION
**Pull request by adamgfraser**

### CODE REVIEW

File: core/shared/src/main/scala/zio/ZIO.scala [hunk 1]
```diff
@@ -1884,6 +1871,22 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
...
+   * this effect.
+   * {{{
+   * readFile("data.json").tapCause(logCause(_))
+   * }}}
+   */
+  final def tapCause[R1 <: R, E1 >: E](f: Cause[E] => ZIO[R1, E1, Any]): ZIO[R1, E1, A] =
+    self.foldCauseM(new ZIO.TapCauseRefailFn(f), ZIO.succeedNow)
+
+  /**
+   * Returns an effect that effect "peeks" at at the defect of this effect.
```
Comment (joprice) [inline core/shared/src/main/scala/zio/ZIO.scala:1885]: typo - `effect that effect` and `at at`

Comment (adamgfraser) [reply core/shared/src/main/scala/zio/ZIO.scala]: Thank you! 🙏

**Review by jdegoes — COMMENTED** [review]
Good to mirror to ZManaged, etc?

**Comment by adamgfraser** [comment]
Yes will do.

**Review by jdegoes — APPROVED** [review]

**Pull request merged**